src/rosdep2/sources_list.py

```diff
@@ -487,9 +487,20 @@ def update_sources_list(sources_list_dir=None, sources_cache_dir=None,
     python_versions = {}
 
     print('Query rosdistro index %s' % get_index_url())
-    for dist_name in sorted(get_index().distributions.keys()):
+    distribution_name_list = sorted(get_index().distributions.keys())
+    if ros_distro not in distribution_name_list:
+        print(
+            'Requested distribution "%s" is not in the index.'
+            ' Ignoring ros_distro argument' % ros_distro)
+        ros_distro = None
```

Comment:
If the user explicitly asks for a specific distro which doesn't exist I would prefer an error message (and failing the invocation) then falling back to do something different.

Reply:
The original implementation aimed to keep original behavior.
Changed to raise `ValueError` if distro doesn't exist https://github.com/ros-infrastructure/rosdep/pull/738/commits/e61e13ff3f97e837444f5036e2bf6c5ab9c26168 

```
root@33270dd54a03:/opt/rosdep# rosdep update --rosdistro foobar
[...]
ERROR: invalid argument value provided:
	Requested distribution "foobar" is not in the index.

```